lib/split/trial.rb

```diff
@@ -125,8 +125,11 @@ def user_experiment_key
       @user_experiment_key ||= @user.alternative_key_for_experiment(@experiment)
     end
 
-    def delete_time_of_assignment_key
-      @user.delete("#{user_experiment_key}:time_of_assignment")
+    def delete_experiment_context_keys
+      keys = @user.all_fields_for_experiment_key(user_experiment_key)
+      keys.each do |key|
+        @user.delete(key) unless key == user_experiment_key || key == Experiment.finished_key(user_experiment_key)
```

Comment:
The finished key is not added at this point. Added the safeguard in case the Split changed when to add the key.